### ISSUE TITLE
Add white/black lists

### ISSUE THREAD

**Issue by aderyabin** [aderyabin/sniffer#29]
Add whitelist/blacklist url template support

```ruby
Sniffer.config do |c|
  c.url_whitelist = /api.github.com/
end

# or
Sniffer.config do |c|
  c.url_blacklist= /localhost:6379/
end
```

**Comment by moofkit** [comment]
@aderyabin have a several questions here.
1. Should whitelist be exclusive? If there are something in whitelist it means other requests are ignored.
```ruby
Sniffer.config.url_whitelist = /api.github.com/

Curl::Easy.http_get("http://localhost:1234/abc") # not logged
Curl::Easy.http_get("https://api.github.com/me") # logged
```

2. Is whitelist rule stronger then blacklist?
```ruby
Sniffer.configurate do |c|
 c.url_whitelist = /api.github.com/
 c.url_blacklist = /github.com/ # this rule will be ignored
end
```

**Comment by aderyabin** [comment]
1. It is  a correct behaviour
2. Yes, too

**Comment by aderyabin** [comment]
Closed by #40
